ui-tests/tools/com.liferay.ide.functional.server/com.liferay.ide.functional.server.start/com.liferay.ide.functional.server.start.tomcat.72/src/com/liferay/ide/functional/server/tests/CustomContextTests.java

```diff
@@ -0,0 +1,92 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.ide.functional.server.tests;
+
+import com.liferay.ide.functional.server.wizard.base.ServerTomcat7xBase;
+import com.liferay.ide.ui.liferay.support.server.PureTomcat72Support;
+import com.liferay.ide.ui.liferay.support.server.ServerSupport;
+
+import java.io.File;
+
+import org.junit.Rule;
+import org.junit.Test;
+
+/**
+ * @author Lily Li
+ */
+public class CustomContextTests extends ServerTomcat7xBase {
+
+	@Test
+	public void addCustomContext() {
+		String folderDir = testServer.getFullServerDir() + "/tomcat-9.0.17/webapps";
```

Comment:
better not use tomcat-9.0.19 here. Please check if we can get tomcet-version from BUndleInfo.java, otherwise, we can add a method to get tomcatDir